src/include/sof/spinlock.h

```diff
@@ -182,43 +143,70 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 #define spin_lock_dbg() do {} while (0)
 #define spin_unlock_dbg() do {} while (0)
 
+static inline int spin_try_lock(spinlock_t *lock)
+{
+	spin_lock_dbg();
+	return arch_try_lock(lock);
+}
+
+#endif
+
 /* all SMP spinlocks need init, nothing todo on UP */
-#define spinlock_init(lock) \
-	arch_spinlock_init(lock)
+static inline void spinlock_init(spinlock_t **lock)
+{
+	arch_spinlock_init(lock);
+#if CONFIG_DEBUG_LOCKS
+	(*lock)->user = __LINE__;
+#endif
```

Comment:
Most likely this won't correctly work with inline functions. I need to check though.

Reply:
Good point, I actually need to figure out a better solution but I have pushed one that partially solves this one (the actual traces might still have the wrong line number as shown in the line number column of `sof-logger`, although the trace messages themselves will show correct lines so where it actually matters the correct line number is passed)